gamemodes/terrortown/gamemode/client/cl_targetid.lua

```diff
@@ -238,11 +238,18 @@ function GM:HUDDrawTargetID()
 	end
 
 	local startpos = client:EyePos()
-
 	local endpos = client:GetAimVector()
+
 	endpos:Mul(MAX_TRACE_LENGTH)
 	endpos:Add(startpos)
 
+	local changedStartpos, changedEndpos = hook.Run("TTTModifyTargetTracedata", startpos, endpos)
```

Comment:
`TTT2ModifyTargetTraceData`

For which scenario is this hook needed? Is it necessary to use startpos and endpos as arguments? Are they needed?

Reply:
@lebroomer wanted a way to modify these for his supersheep

Reply:
Okay, and do he need these vars? It's totally okay if it's needed, then he doesn't need to recalculate. But if it's not needed, there should be as few arguments as possible

Reply:
I don't think he needs them. But why don't give the user the veraibles that are available? E.g. for a simple shift

Reply:
That argument won't count, imagine giving any data you have in TTTBeginRound hook etc :D

Reply:
I thought these values are useful here. But I can remove them

Reply:
I don't think this hook is necessary. I really like the ```GM:TTTModifyTargetedEntity(ent, distance)``` hook. This can be used for addons like the supersheep too by simply calculating a traceline in the addon and inserting the information, which are calculated there.

Reply:
OKay, then I'll remove this one completely since I don't need it as well :D